go/cmd/dolt/commands/tblcmds/import.go

```diff
@@ -724,26 +725,47 @@ func NameAndTypeTransform(row sql.Row, rowOperationSchema sql.PrimaryKeySchema,
 	row = applyMapperToRow(row, rowOperationSchema, rdSchema, nameMapper)
 
 	for i, col := range rowOperationSchema.Schema {
-		switch col.Type {
-		case sql.Boolean, sql.Int8, sql.MustCreateBitType(1): // TODO: noms bool wraps MustCreateBitType
-			switch row[i].(type) {
-			case int8:
-				val, ok := stringToBoolean(strconv.Itoa(int(row[i].(int8))))
-				if ok {
-					row[i] = val
-				}
-			case string:
-				val, ok := stringToBoolean(row[i].(string))
-				if ok {
-					row[i] = val
-				}
-			case bool:
-				row[i] = row[i].(bool)
-			}
+		// Check if this a string that can be converted to a boolean
+		val, ok := detectAndConvertToBoolean(row[i], col.Type)
+		if ok {
+			row[i] = val
+			continue
 		}
 
 		switch col.Type.(type) {
 		case sql.StringType:
```

Comment:
I know this line didn't change, but it still looked odd... do we need this, or should string types get picked up in the default case?  i.e. do we want string types to have empty strings converted to nils?  I suspect we may want to leave any literal empty string string types as empty string, but it would be good to add a short comment if that's the intent to make it explicit.

Reply:
Yea I can add a comment